### PR TITLE
Add integration tests to the CICD pipeline

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Description

See `integration/README.md`
Note: Not to be merged yet as we need to add the required env variables to the `clerkinc/javascript` repo. 

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [x] `build/tooling/chore`

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
+name: Quality checks and tests
+on:
+  push:
+  pull_request:
+    branches: [ main, master ]
```
Comment (nikosdouvlis) [inline .github/workflows/checks.yml:5]: Note: some of these are intentionally left as-is for testing purposes and will be removed before merging.

File: integration/presets/longRunningApps.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import { remix } from './remix';
+
+/**
+ * A list of long-running applications that can be used in tests.
+ * These are applications that are started once and then used for all tests,
+ * making the tests run faster as the app doesn't need to be started for each test.
+ */
+export const createLongRunningApps = () => {
+  const configs = [
+    { id: 'react.vite.withEmailCodes', config: react.vite, env: envs.withEmailCodes },
```
Comment (anagstef) [inline integration/presets/longRunningApps.ts:15]: ❓ Is this `id` just a string, or is it matching something?

Comment (nikosdouvlis) [reply integration/presets/longRunningApps.ts:15]: Just a string identifier you can use to boot a group of apps or a specific app, eg:

```
E2E_APP_ID=react.vite.* npm run test:integration:base -- --grep @generic
```

File: integration/constants.ts [hunk 3]
```diff
@@ -1,6 +1,13 @@
...
 
 export const constants = {
   TMP_DIR: path.join(process.cwd(), '.temp_integration'),
   APPS_STATE_FILE: path.join(process.cwd(), '.temp_integration', 'state.json'),
+  E2E_APP_URL: process.env.E2E_APP_URL,
+  E2E_APP_ID: process.env.E2E_APP_ID,
+  CLEANUP: !(process.env.CLEANUP === '0' || process.env.CLEANUP === 'false'),
+  DEBUG: process.env.DEBUG === 'true' || process.env.DEBUG === '1',
+  E2E_APP_PK: process.env.E2E_APP_PK,
+  E2E_APP_SK: process.env.E2E_APP_SK,
```
Comment (anagstef) [inline integration/constants.ts:12]: ❓ Can we add an explanation for each of these constants?

**Review by anagstef — APPROVED** [review]
Amazing job @nikosdouvlis!

**Review by desiprisg — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**